react-components/src/components/RevealContext/RevealContext.tsx

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { Cognite3DViewer, type Cognite3DViewerOptions } from '@cognite/reveal';
+import { type CogniteClient } from '@cognite/sdk/dist/src';
+import { type ReactNode, useEffect, useMemo, useRef, useState } from 'react';
+import { type Color } from 'three';
+import { I18nContextProvider } from '../i18n/I18n';
+import { ViewerContext } from '../RevealCanvas/ViewerContext';
+import { NodeCacheProvider } from '../NodeCacheProvider/NodeCacheProvider';
+import { AssetMappingCacheProvider } from '../NodeCacheProvider/AssetMappingCacheProvider';
+import { PointCloudAnnotationCacheProvider } from '../NodeCacheProvider/PointCloudAnnotationCacheProvider';
+import { Reveal3DResourcesCountContextProvider } from '../Reveal3DResources/Reveal3DResourcesCountContext';
+import { SDKProvider } from '../RevealCanvas/SDKProvider';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { useRevealKeepAlive } from '../RevealKeepAlive/RevealKeepAliveContext';
+
+export type RevealContextProps = {
+  color?: Color;
+  sdk: CogniteClient;
+  appLanguage?: string;
+  children?: ReactNode;
+  viewerOptions?: Pick<
+    Cognite3DViewerOptions,
+    | 'antiAliasingHint'
+    | 'loadingIndicatorStyle'
+    | 'rendererResolutionThreshold'
+    | 'antiAliasingHint'
+    | 'ssaoQualityHint'
+    | 'pointCloudEffects'
+    | 'enableEdges'
+    | 'useFlexibleCameraManager'
+  >;
+};
+
+export const RevealContext = (props: RevealContextProps): ReactNode => {
+  const viewer = useRevealFromKeepAlive(props);
+
+  const queryClient = useMemo(() => {
+    return new QueryClient();
+  }, []);
+
+  if (viewer === undefined) return <></>;
```

Comment:
This should be moved above `useMemo()`

Reply:
Hooks cannot be called after conditional return statements. https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level